lighthouse-core/test/audits/installable-manifest-test.js

```diff
@@ -85,97 +91,125 @@ describe('PWA: webapp install banner audit', () => {
+      const artifacts = generateMockArtifacts();
+      artifacts.InstallabilityErrors.errors.push({errorId: 'manifest-missing-suitable-icon',
+        errorArguments: []});
+      const context = generateMockAuditContext();
+
+      return InstallableManifestAudit.audit(artifacts, context).then(result => {
+        assert.strictEqual(result.score, 0);
+        const items = result.details.items;
+        assert.ok(items[0].reason.includes('number of arguments'));
+      });
+    });
+
+    it('fails when we receive an unknown InstallabilityError id', () => {
+      const artifacts = generateMockArtifacts();
+      artifacts.InstallabilityErrors.errors.push({errorId: 'new-error-id', errorArguments: []});
       const context = generateMockAuditContext();
 
       return InstallableManifestAudit.audit(artifacts, context).then(result => {
         assert.strictEqual(result.score, 0);
-        assert.ok(result.explanation.includes('failed to be fetched'), result.explanation);
+        const items = result.details.items;
+        assert.ok(items[0].reason.formattedDefault.includes('is not recognized'));
+      });
+    });
+  });
 
-        const details = result.details.items[0];
-        assert.strictEqual(details.failures.length, 1, details.failures);
-        assert.strictEqual(details.hasStartUrl, true);
-        assert.strictEqual(details.hasIconsAtLeast144px, true);
+  describe('warnings', () => {
+    it('presents a warning if a warn-not-offline-capable if present', () => {
+      const artifacts = generateMockArtifacts(manifestDirtyJpgSrc);
+      artifacts.InstallabilityErrors.errors.push({errorId: 'warn-not-offline-capable'});
+      const context = generateMockAuditContext();
+
+      return InstallableManifestAudit.audit(artifacts, context).then(result => {
+        expect(result.warnings).toHaveLength(1);
+        expect(result.warnings[0].formattedDefault).toMatch('not work offline');
+        expect(result.warnings[0].formattedDefault).toMatch('August 2021');
       });
     });
   });
 
   it('fails if icons were present, but no valid PNG present', () => {
     const artifacts = generateMockArtifacts(manifestDirtyJpgSrc);
+    artifacts.InstallabilityErrors.errors.push({errorId: 'manifest-missing-suitable-icon',
+      errorArguments: [{name: 'minimum-icon-size-in-pixels', value: '144'}]});
     const context = generateMockAuditContext();
 
     return InstallableManifestAudit.audit(artifacts, context).then(result => {
       assert.strictEqual(result.score, 0);
-      assert.ok(result.explanation.includes('PNG icon'), result.explanation);
-
-      const details = result.details.items[0];
-      assert.strictEqual(details.failures.length, 1, details.failures);
-      assert.strictEqual(details.hasStartUrl, true);
-      assert.strictEqual(details.hasIconsAtLeast144px, false);
+      const items = result.details.items;
+      assert.ok(items[0].reason.formattedDefault.includes('PNG'));
```

Comment:
```suggestion
      expect(items[0].reason).toBeDisplayString(/PNG/);
```

Reply:
for these new tests, consider testing more of the string, as well, instead of just a short substring. e.g.
```suggestion
      expect(items[0].reason).toBeDisplayString(/^Manifest does not contain a suitable icon - PNG, SVG or WebP format of at least 144 px is required/);
```
would need a max-len suppression, but gives a lot more context to the test reader _and_ (in this case) tests that the number replacement is working :)